### PR TITLE
📖 Use client.New instead of manager client in tests

### LINKED ISSUE #2018: Don't recommend using manager.GetClient in tests
Using the `manager.GetClient()` in tests is a bad idea and should not be recommended by the documentation. The manager's client is a cached `client.Reader` which is fine for a controller - but in tests you probably want a strongly consistent client.

This is the documentation i am referring to:
https://github.com/kubernetes-sigs/kubebuilder/blame/master/docs/book/src/cronjob-tutorial/testdata/project/controllers/suite_test.go#L115-L116

There's an older issue that explains the problem in more detail: https://github.com/kubernetes-sigs/controller-runtime/issues/343

I'd suggest updating the documentation about that. What do you think? Did anybody else run into that issue yet?

/kind bug
/kind documentation

### PR DESCRIPTION
**Pull request by leovct**
**Description**
This PR improves the testing section of the documentation.
The idea is to recommend to use a strongly consistent client ```client.New``` instead of the manager's client in the case of tests.

**Motivation**
* Closes: #2018
* It is highly inspired by this discussion https://github.com/kubernetes-sigs/controller-runtime/issues/343.

### CODE REVIEW

**Comment by leovct** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
>  **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

Done!

File: docs/book/src/cronjob-tutorial/testdata/project/controllers/suite_test.go [hunk 1]
```diff
@@ -132,10 +139,6 @@ var _ = BeforeSuite(func() {
 		err = k8sManager.Start(ctrl.SetupSignalHandler())
 		Expect(err).ToNot(HaveOccurred())
 	}()
-
-	k8sClient = k8sManager.GetClient()
-	Expect(k8sClient).ToNot(BeNil())
```
Comment (camilamacedo86) [inline docs/book/src/cronjob-tutorial/testdata/project/controllers/suite_test.go:137]: imo; we should do it as we do in the scaffolds (k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})). See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/controllers/suite_test.go

Could you please update it for it be compatible with how we scaffold it?

Comment (leovct) [reply docs/book/src/cronjob-tutorial/testdata/project/controllers/suite_test.go:137]: Ok, so should I delete the whole manager part (from line 105 to line 141) and add my changes in the comment on line 99?

Comment (Adirio) [reply docs/book/src/cronjob-tutorial/testdata/project/controllers/suite_test.go:137]: @camilamacedo86 it is exactly done as described: `8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})` in line 101 of this same file. It is just hidden by GitHub.

**Comment by Adirio** [comment]
/ok-to-test
/lgtm
/approve
/hold

Holding until Camila can answer the above comment, but I think the suggestion she made is already applied.

**Review by Adirio — APPROVED** [review]

**Review by irvifa — APPROVED** [review]

**Comment by camilamacedo86** [comment]
/test pull-kubebuilder-e2e-k8s-1-16-15
--

**Comment by DirectXMan12** [comment]
Follow up: this breaks the tests, and is actually wrong -- the advice is to not use the manager's client in *assertions* in tests.  You should always use the manager's client in the reconciler if that's what you're also using in production.

**Pull request merged**